ddtrace/propagation/http.py

```diff
@@ -915,8 +963,19 @@ def _extract_configured_contexts_avail(normalized_headers):
     def _resolve_contexts(contexts, styles_w_ctx, normalized_headers):
         primary_context = contexts[0]
         links = []
+
         for context in contexts[1:]:
             style_w_ctx = styles_w_ctx[contexts.index(context)]
+
+            # special case where baggage is first in propagation styles
+            # why would you do that? I don't know, but it's possible
+            if style_w_ctx[0] == _PROPAGATION_STYLE_BAGGAGE:
+                baggage_context = contexts[0]
+                contexts.append(baggage_context)
+                del contexts[0]
+                styles_w_ctx.append(_PROPAGATION_STYLE_BAGGAGE)
+                del style_w_ctx[0]
```

Comment:
Should just short circuit here? 
```suggestion
                continue
```